yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -195,3 +199,46 @@ func (d *RangeTableFuncDetector) GetIssues() []QueryIssue {
 	}
 	return issues
 }
+
+type SelectStmtDetector struct {
+	query                       string
+	limitOptionWithTiesDetected bool
+}
+
+func NewSelectStmtDetector(query string) *SelectStmtDetector {
+	return &SelectStmtDetector{
+		query: query,
+	}
+}
+
+func (d *SelectStmtDetector) getSelectStmtFromProto(msg protoreflect.Message) (*pg_query.SelectStmt, error) {
+	protoMsg := msg.Interface().(proto.Message)
+	selectStmtNode, ok := protoMsg.(*pg_query.SelectStmt)
+	if !ok {
+		return nil, fmt.Errorf("failed to cast %s to %s", queryparser.PG_QUERY_SELECTSTMT_NODE, queryparser.PG_QUERY_SELECTSTMT_NODE)
+	}
+	return selectStmtNode, nil
+}
+
+func (d *SelectStmtDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_SELECTSTMT_NODE {
+		selectStmtNode, err := d.getSelectStmtFromProto(msg)
+		if err != nil {
+			return err
+		}
+		// checks if a SelectStmt node uses a FETCH clause with TIES
+		// https://www.postgresql.org/docs/13/sql-select.html#SQL-LIMIT
+		if selectStmtNode.LimitOption == pg_query.LimitOption_LIMIT_OPTION_WITH_TIES {
```

Comment:
define this limit option const `pg_query.LimitOption_LIMIT_OPTION_WITH_TIES` in `queryparser` 
After these two changes (this and the above one), you won't need to use the `pg_query` package. Because we already have `queryparser` pkg for parser related stuff. So should we not use the `pg_query` directly here.

Reply:
Got it. Don't think we can completely avoid this because the return type would be a pg_query.SelectStmt. But we probably don't need to have an explicit dependency. 

But broadly speaking, agree with not preferring to use pg_query directly here. (as much as possilble).

Reply:
Yes, as much as possible we should avoid using it directly in the queryissue.